packages/plugin/src/object.ts

```diff
@@ -5,10 +5,74 @@ import {
   StoreDeserializeHelpers,
   StoreSerializeHelpers,
   StateDescriptorReturnType,
-  StateDescriptorsSerializedType
+  StateDescriptorsSerializedType,
+  StateType,
+  StateTypesSerializedType,
+  StateTypesValueType,
+  StateTypesReturnType,
+  StateTypeReturnType
 } from '@edtr-io/abstract-plugin-state'
 import * as R from 'ramda'
 
+export function newObject<Ds extends Record<string, StateType>>(
+  types: Ds
+): StateType<
+  StateTypesSerializedType<Ds>,
+  StateTypesValueType<Ds>,
+  { __types: StateTypesReturnType<Ds> } & StateTypesReturnType<Ds>
```

Comment:
`StateTypesReturnType` doesn't work correctly atm. In the tests, it resolves to `any`...

Reply:
Fixed